sound/soc/sof/intel/cnl.c

```diff
@@ -294,6 +294,9 @@ const struct snd_sof_dsp_ops sof_cnl_ops = {
 	.pre_fw_run = hda_dsp_pre_fw_run,
 	.post_fw_run = hda_dsp_post_fw_run,
 
+	/* parse platform specific extended manifest */
+	.parse_platform_ext_manifest = hda_dsp_ext_man_get_cavs_config_data,
```

Comment:
Why add this parse as ops?
Will we have different type of ext_manifest for different platform?
I think you can make something same with https://github.com/thesofproject/linux/pull/2459

Reply:
@xiulipan yes we will. For example, why would the IMX platforms have this elem?

Reply:
@ranj063 A good question here, but I think we should parse the platform manifest in one place and config them to each platform.
Otherwise we will need to add a almost same function for parse in each platform.

Reply:
@xiulipan I'm afraid this is a bit wrong audience. The ABI was already added to FW with 3.17 in https://github.com/thesofproject/sof/pull/2893
It's possible to just ignore this and agree a new interface to pass clock information in non-hw-specific way, but this needs to be raised in FW.

Reply:
@ranj063 @fredoh9 @kv2019i My only concern here is about the generic part in `hda_dsp_ext_man_get_cavs_config_data`, if we need implement this for each platform there will be too many duplicate code. I will suggest to only wrap specific handler to platform and fallback to unknown handler to unsupported platforms.